smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/KafkaProducer.java

```diff
@@ -80,6 +85,40 @@
     @CheckReturnValue
     Uni<List<PartitionInfo>> partitionsFor(String topic);
 
+    /**
+     * @return the Uni emitting {@code null} when the init transactions executes.
+     */
+    @CheckReturnValue
+    Uni<Void> initTransactions();
+
+    /**
+     * @return the Uni emitting {@code null} when the begin transactions executes.
+     */
+    @CheckReturnValue
+    Uni<Void> beginTransaction();
+
+    /**
+     * @return the Uni emitting {@code null} when the commit transactions executes.
+     */
+    @CheckReturnValue
+    Uni<Void> commitTransaction();
+
+    /**
+     * @return the Uni emitting {@code null} when the abort transactions executes.
+     */
+    @CheckReturnValue
+    Uni<Void> abortTransaction();
+
+    /**
+     *
+     * @param offsets topic partition offsets to commit into transaction
+     * @param groupMetadata consumer group metadata of the exactly-once consumer
+     * @return the Uni emitting {@code null} when the init transactions executes.
```

Comment:
I'm not sure when exactly the returned `Uni` completes, but it sure isn't when "init transactions executes" :-)

Reply:
:) true